contracts/adapters/HegicAdapter.sol

```diff
@@ -288,6 +289,25 @@ contract HegicAdapter is IProtocolAdapter {
         emit Exercised(account, optionsAddress, optionID, amount, profit);
     }
 
+    /**
+     * @notice Function to bulk claim rHEGIC2 rewards from liquidity utilization
+     * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
+     * @param optionIDs is an array of all the option ids we want to claim rewards for
+     */
+    function claimRewards(address rewardsAddress, uint256[] calldata optionIDs)
+        external
+        returns (uint256)
+    {
+        IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+        uint256 i = 0;
+        uint256 j = 0;
+        while (i < optionIDs.length) {
+          try rewardsContract.getReward(optionIDs[i]) {} catch {j+=1;}
```

Comment:
I'd prefer if this reverts instead of continuing execution, it's easier to keep track of which option IDs have been claimed that way.

For example, if optionIDs from the range 100 - 300 fails, you just retry the whole range, instead of retrying a few option IDs in there.